course_contents/2_intro_to_python/notes/02. Loops.md

```diff
@@ -0,0 +1,159 @@
+### Loops
+- loops are used to run blocks of code multiple times
+- there are two kinds of loops in Python:
+    1. while loops
+    2. for loops
+
+**Extra resources**
+[What is a loop?](https://python.tecladocode.com/3_continuous_countries/1_loops.html#what-is-a-loop)
+
+### The `while` loop
+- while loop runs an indefinite number of times i.e. while loop runs until the given conditional becomes False
+- syntax for writing the while loop:
+    1. use the `while` keyword at the beginning of the loop
+    2. the keyword should be followed with a correctly written conditional with a colon at the end of it
+    3. indented block of code that will run until the loop breaks
+
+An example of the properly written `while` loop:
+
+```py
+counter = 10
+while counter > 0:
+  print(f"{counter} seconds left until New Year!")
+  counter -= 1
+print("Happy New Year!")
+```
+
+An example of the `while` loop that will run until the program is manually stopped:
+
+```py
+counter = 10
+while counter > 0: # the counter never gets changed, causing the conditional to be always True
+  print(f"{counter} seconds left until New Year!")
+```
+
+### The `for` loop
+- for loop runs a given number of times
+- general syntax for `for` loop is:
+
+```py
+for variable in iterable:
+    block_of_code
+```
+- an iterable is an object that can be iterated over e.g. a list
+- an object is iterable if it can return the iterator object (more on that in the OOP section)
```

Comment:
I don't know if this line brings any value here, and students don't know what OOP is yet.

Reply:
I agree that we can remove it if we don't mind mentioning iterables as "things" rather than objects which I am completely okay with if Jose wants us to go that way

Reply:
I agree that mentioning iterators here might be too complicated. Instead of "things" though we could say "collections" ?

Reply:
I removed the line about iterators, although I left the term iterable in the text. I've used the term collection at a couple of places but decided to change it to iterables. The reason for that is because I think we should incorporate terms like that into students' vocabulary even though they might not know about iterators yet. 
I agree that a collection would be a nice substitute for iterable and I'm happy to change it to that if you think that is the better option

Reply:
I think saying "iterable" here with the explanation you've given is good too, I'm happy with that if you prefer that term. It is more accurate and I think it's not too confusing!